File: source/javaclient/com/ib/client/WshEventData.java
/* Copyright (C) 2022 Interactive Brokers LLC. All rights reserved. This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

package com.ib.client;

public class WshEventData {

    private int m_conId;
    private String m_filter;
    private boolean m_fillWatchlist;
    private boolean m_fillPortfolio;
    private boolean m_fillCompetitors;

    public int conId() { return m_conId; }
    public String filter() { return m_filter; }
    public boolean fillWatchlist() { return m_fillWatchlist; }
    public boolean fillPortfolio() { return m_fillPortfolio; }
    public boolean fillCompetitors() { return m_fillCompetitors; }

    public WshEventData(int conId) {
        m_conId = conId;
        m_filter = "";
        m_fillWatchlist = false;
        m_fillPortfolio = false;
        m_fillCompetitors = false;
    }

    public WshEventData(String filter, boolean fillWatchlist, boolean fillPortfolio, boolean fillCompetitors) {
        m_conId = Integer.MAX_VALUE;
        m_filter = filter;
        m_fillWatchlist = fillWatchlist;
        m_fillPortfolio = fillPortfolio;
        m_fillCompetitors = fillCompetitors;
    }

}

File: source/csharpclient/client/WshEventData.cs
/* Copyright (C) 2022 Interactive Brokers LLC. All rights reserved. This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

namespace IBApi
{
    public class WshEventData
    {
        public int ConId { get; set; }

        public string Filter { get; set; }

        public bool FillWatchlist { get; set; }

        public bool FillPortfolio { get; set; }

        public bool FillCompetitors { get; set; }

        public WshEventData()
        {
            ConId = int.MaxValue;
            Filter = Order.EMPTY_STR;
            FillWatchlist = false;
            FillPortfolio = false;
            FillCompetitors = false;
        }

        public WshEventData(int conId)
        {
            ConId = conId;
            Filter = Order.EMPTY_STR;
            FillWatchlist = false;
            FillPortfolio = false;
            FillCompetitors = false;
        }

        public WshEventData(string filter, bool fillWatchlist, bool fillPortfolio, bool fillCompetitors)
        {
            ConId = int.MaxValue;
            Filter = filter;
            FillWatchlist = fillWatchlist;
            FillPortfolio = fillPortfolio;
            FillCompetitors = fillCompetitors;
        }
    }
}
